modin/pandas/base.py

```diff
@@ -783,6 +820,7 @@ def apply(
         args=(),
         **kwds,
     ):
+        self._validate_function(func, on_invalid=AssertionError)
```

Comment:
1.3.0 brought a bit of inconsistency in error types, `apply` and `aggregate` now [raise an `AssertionError`](https://github.com/pandas-dev/pandas/pull/38867/files#diff-2257b34410aee27eb14e348b9545fef2e212ff93bd72af02d700ae8df43d97bbR350) in case of an invalid function, when for `transform` [it's still `TypeError/ValueError`](https://github.com/pandas-dev/pandas/pull/38867/files#diff-2257b34410aee27eb14e348b9545fef2e212ff93bd72af02d700ae8df43d97bbR350), that's why this commit introduces `_validate_function` which can raise different types of exceptions